packager/media/base/aes_cryptor.cc

```diff
@@ -30,20 +26,21 @@ namespace shaka {
 namespace media {
 
 AesCryptor::AesCryptor(ConstantIvFlag constant_iv_flag)
-    : aes_key_(new AES_KEY),
-      constant_iv_flag_(constant_iv_flag),
-      num_crypt_bytes_(0) {
-  CRYPTO_library_init();
+    : constant_iv_flag_(constant_iv_flag), num_crypt_bytes_(0) {
+  mbedtls_cipher_init(&cipher_ctx_);
 }
 
-AesCryptor::~AesCryptor() {}
+AesCryptor::~AesCryptor() {
+  mbedtls_cipher_free(&cipher_ctx_);
+}
 
 bool AesCryptor::Crypt(const std::vector<uint8_t>& text,
                        std::vector<uint8_t>* crypt_text) {
   // Save text size to make it work for in-place conversion, since the
   // next statement will update the text size.
   const size_t text_size = text.size();
-  crypt_text->resize(text_size + NumPaddingBytes(text_size));
+  // mbedtls requires an extra block's worth of output buffer available.
```

Comment:
Is it a bug? Is it mentioned in the mbedtls documentation?

I glanced through the mbedtls_cipher_crypt implementation quickly but I don't see that an extra block is needed for ECB and CBC modes. (But I may have overlooked it)

Also, I am not sure if it works everywhere it is used. It might not always be possible to require the output size to be larger than the input size (especially for in-place encryption / decryption).

Reply:
It is mentioned in the documentation.  I overlooked it initially, but after valgrind reported overflows, I re-read the docs and found it there.

https://github.com/Mbed-TLS/mbedtls/blob/55fd0b9fc1a2a8e0720c8ab15e06cea579aa4f4b/include/mbedtls/cipher.h#L1035-L1037

```
 * \param output        The buffer for the output data. This must be able to
 *                      hold at least `ilen + block_size`. 
```

Reply:
I see. Thanks for confirming! It also says that "This must not be the same buffer as \p input.", which I assume it means that it does not support in-place encryption/decryption. I think we do use in-place encryption / decryption in a few places.

Reply:
We do, but I've been able to verify that they still work.  Some did not, and were replaced.  I believe it depends on the cipher mode in practice, even though the header docs state that it is not supported in general.  All the tests work, and valgrind is happy.